projects/backend/src/main/java/backend/security/JwtAuthenticationFilter.java

```diff
@@ -0,0 +1,68 @@
+package backend.security;
+
+import static com.auth0.jwt.algorithms.Algorithm.HMAC512;
+
+import com.auth0.jwt.JWT;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Date;
+
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.AuthenticationException;
+import org.springframework.security.core.userdetails.User;
+import org.springframework.security.web.authentication.UsernamePasswordAuthenticationFilter;
+
+
+
+public class JwtAuthenticationFilter extends UsernamePasswordAuthenticationFilter {
+  private AuthenticationManager authenticationManager;
+
+  public JwtAuthenticationFilter(AuthenticationManager authenticationManager) {
+    this.authenticationManager = authenticationManager;
+  }
+
+  @Override
+  public Authentication attemptAuthentication(
+                                              HttpServletRequest req,
+                                              HttpServletResponse res)
+                                              throws AuthenticationException {
+    try {
+      backend.domain.User creds = new ObjectMapper()
+          .readValue(req.getInputStream(), backend.domain.User.class);
+
+      return authenticationManager.authenticate(
+          new UsernamePasswordAuthenticationToken(
+              creds.getIndexNumber(),
+              creds.getPassword(),
+              new ArrayList<>())
+      );
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
```

Comment:
How does throwing ```RuntimException``` influences on our server execution? Does it stop only session of a current user or does it kill an entire application? I think that we should also log that kind of event, You can use ```log4j``` or ```sl4fj``` (it's a facade that allows to switch various loggers)